options.go

```diff
@@ -69,12 +71,9 @@ func SMSSenderID(senderID string) SendSMSOption {
 	})
 }
 
-type payloadUpdater interface {
-	updatePayload(payload) payload
-}
-
-type CommonOption struct {
-	payloadUpdater
+type CommonOption interface {
```

Comment:
@andrewoh531 it's not clear why this changed? `CommonOption` now appears to be a publicly exported interface that contains two very specific interfaces.

Reply:
I might need to lean on @dagoof here as he helped me simplify this. I think the changes were to reduce the complexity.

Reply:
@dagoof you happy here?

Reply:
I'm happier with this than how it was; but I think there's still issues with what's going on here.

It's not exposed that `Personalisation` satisfies the `CommonOption` interface because we don't know what methods it takes to support the interface. Similarly, it's not totally clear that a `CommonOption` can be used as an argument to `SendSMS` which accepts `SendSMSOptions`s.

And even then, it's starting to seem unlikely that this approach scales beyond one or two functions. Do we need to create a `GetTemplateOption` interface, `GenerateTemplatePreviewOption` interface? for all methods that accept optional parameters.

We'll need to mint a new option interface (or multiple) for every exposed function; as well as constructors for all possible options for those interfaces. We'll end up with something like this for every single function:

```
type GetAllTemplatesOption interface { ... }

func TemplateType(string) GetAllTemplatesOption { ... }

// ... and more if other options are available

func GetAllTemplates(...GetAllTemplatesOption) TemplateList { ... }
```


I'm really not sure on the tradeoff. Sure we can very specifically describe what's available to each function in a type-safe and semantic way without any ambiguity. It takes a bit more work on our end but I don't mind that.

I'm just not sure if it ends up being any clearer to the user about how to use the thing over a plain `type GetAllTemplatesOptions struct { TemplateType string }` option struct

Reply:
Yeah not too sure without sitting on it for a bit and trying out some things. Are you both happy with the addition of an exported `PersonalisationOption` interface that we'll add with this PR?

Reply:
I'll defer to @dagoof